runtime/vm_function_test.go

```diff
@@ -12,6 +12,14 @@ func TestFunction(t *testing.T) {
 	expect(t, `f1 := func() {}; f2 := func() { return f1(); }; f1(); out = f2();`, nil, objects.UndefinedValue)
 	expect(t, `f := func(x) { x; }; out = f(5);`, nil, objects.UndefinedValue)
 
+	expect(t, `f := func(...x) { return x; }; out = f(1,2,3);`, nil, &objects.Array{Value: []objects.Object{
```

Comment:
It should be possible to have functions with multiple params with the last parameter being variadic.

For example `func(a, ...b)` should be valid.

Reply:
I fully agree, just wanted to get feedback on the overall approach before I added more complexity. Once we hash out the performance concerns I'll make this change. Should only be a matter of consolidating the last X variables instead of all of them. Parsing-wise, shouldn't add too much complexity. All the magic happens in the `compiler`'s `parseIdentList` function